runtime/vm/MHInterpreter.cpp

```diff
@@ -555,6 +555,13 @@ VM_MHInterpreter::dispatchLoop(j9object_t methodHandle)
 			/* Get MethodHandle for this operation from the VarHandle's handleTable */
 			j9object_t handleTable = J9VMJAVALANGINVOKEVARHANDLE_HANDLETABLE(_currentThread, varHandle);
 			j9object_t methodHandleFromTable = J9JAVAARRAYOFOBJECT_LOAD(_currentThread, handleTable, operation);
+
+			if (NULL == methodHandleFromTable) {
+				prepareForExceptionThrow(_currentThread);
+				setCurrentExceptionUTF(_currentThread, J9VMCONSTANTPOOL_JAVALANGUNSUPPORTEDOPERATIONEXCEPTION, NULL);
```

Comment:
Does this need to use the vmconstantpool?  Can we look up the class here if it's needed as it will only be used in error cases? 

I prefer not to grow the vmconstantpool if we can avoid it as every class in the pool is loaded at startup.

Reply:
I have not noticed `setCurrentException*` examples without `J9VMCONSTANTPOOL_*` macros. So, vmconstantpool is required to throw the `UnsupportedOperationException` from `setCurrentException*`.

Reply:
I also observed that JEP370 testcases catch and ignore `UnsupportedOperationException`. We will under-perform in such scenarios if we look up for the exception class.

Reply:
I would rather we lookup the class, allocate an instance, and then throw with `VM_VMHelpers::setExceptionPending` than putting the class in the vm CP.

Test cases will of course trigger the `UnsupportedOperationException`, real applications won't.  I'd rather optimize for real apps then tests :)

Reply:
Updated to look up the class and using `setExceptionPending` to throw `UnsupportedOperationException`.